### PR TITLE
CVE-2016-1640 and CVE-2011-3965

### PR DESCRIPTION
**Pull request by hkdahal**
This PR explores CVE-2016-1640, inline extension installation dialog bug, and CVE-2011-3965, a NULL pointer dereference problem in safe_browsing.

### CODE REVIEW

File: cves/CVE-2011-3965.yml [hunk 1]
```diff
@@ -94,20 +103,25 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer:
-  date:
-  automated:
-  google:
-  contest:
+  answer: Some person tried the beta version of the chrome on windows, and their browser crashed.
```
Comment (andymeneely) [inline cves/CVE-2011-3965.yml]: This is the reason your YML isn't building. Needs to be a multiline string, using the pipe char

File: cves/CVE-2011-3965.yml [hunk 2]
```diff
@@ -32,7 +32,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  Google Chrome crashed when the program tried to call CheckSignature() function.
```
Comment (bxy3204) [inline cves/CVE-2011-3965.yml]: Consider removing Chromium-specific references such as the function name CheckSignature() from the description.

File: cves/CVE-2011-3965.yml [hunk 3]
```diff
@@ -117,7 +133,10 @@ interesting_commits:
     emerging themes?
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    Due to the nature of this vulnerability, there were no any interesting commits in between. There are no commits to this file
+    in between the fix commit and the VCC. The vulnerability lived for around 2 months. It was interesting to see that the person who reported
+    the vulnerability knew what was going and the fix was made right away and the fix was simply around 4 lines of code.
```
Comment (bentgarbacz) [inline cves/CVE-2011-3965.yml]: The last sentence is hard to read, you should edit it to make it more understandable.

File: cves/CVE-2011-3965.yml [hunk 4]
```diff
@@ -197,4 +220,10 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    NULL Pointer Dereference is one of the commonly seen vulnerabilities.
+    It is very important to validate data received from outside the module
+    before acting upon it. Sanity-checking before using could nearly prevent all
+    NULL pointer dereference problems. However, it is impossible to sanity-check
+    all the data all the times. Thus, writing some strong unit tests to check such
```
Comment (bentgarbacz) [inline cves/CVE-2011-3965.yml]: "all the data all the times" > "all data all the time"

File: cves/CVE-2016-1640.yml [hunk 5]
```diff
@@ -32,7 +32,17 @@ description_instructions: |
...
+  When a chrome user tries to install an extension from a website, that
+  website could redirect to another website and still show the extension installation
+  dialogue started from previous website.
+  For example, a crafted website could show a link to install an official
+  extension from some service say Google, and when user clicks on it, the chrome
+  will take you to the Google support page and will be prompted with an extension
+  installation dialogue initiated by the crafted website. The presented extension
+  could be a fake extension intended to be an attack. The user would think it's official
+  one given that user is on the official support website. The dialog doesn't show
+  the origin, so it gives even more credibility to the attack.
```
Comment (bentgarbacz) [inline cves/CVE-2016-1640.yml]: Can you add a brief explanation about what caused this issue in the description?

File: cves/CVE-2016-1640.yml [hunk 6]
```diff
@@ -81,9 +95,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    The original code was tested with many browser UI test. The fix also
```
Comment (bentgarbacz) [inline cves/CVE-2016-1640.yml]: "browser UI test" > "browser UI tests"

File: cves/CVE-2011-3965.yml (hunk 3 above)
Comment (fernflores0463) [reply cves/CVE-2011-3965.yml]: Even if there were no interesting commits, a general overview of the small changes that were made should be included.

File: cves/CVE-2016-1640.yml [hunk 7]
```diff
@@ -202,4 +241,16 @@ mistakes:
...
+    could have been implemented that way. However, I can understand it's impossible
+    to think about all the scenarios on requirements. So, that's why if proper design
+    decisions were made regarding how to handle such scenario when user navigates to different
+    page or when user is redirected to different page.
+    Based on bug report, it seems like they had made better design decision for next version of Chrome.
+    This vulnerability needed medium-level security attention because the install dialog
+    didn't show the origin and users could be tricked into installing malicious extensions. So, the initial fix seemed like a hack,
+    where they were simply trying to mitigate this problem for now. Second fix improved the previous fix however.
+    The final solution for this was to redesign the architecture of how the dialog is shown. A comment on the bug report indicated that
+    they were planning to redesign the dialog, so looks like they knew the proper mitigation for this problem.
```
Comment (fernflores0463) [inline cves/CVE-2016-1640.yml]: Cool, I like how you explained the problem, the temporary solution, and the more permanent solution that is going to be released in the future

File: cves/CVE-2011-3965.yml (hunk 2 above)
Comment (bxy3204) [reply cves/CVE-2011-3965.yml]: Check the instructions, chromium-specific stuff should be removed. As an outsider to chromium, the sentence "Google Chrome crashed when the program tried to call CheckSignature() function." means nothing to me.

Comment (hkdahal) [reply cves/CVE-2011-3965.yml]: done!

File: cves/CVE-2011-3965.yml (hunk 3 above)
Comment (hkdahal) [reply cves/CVE-2011-3965.yml]: fixed the last sentence, and found a commit that seemed interesting to me.

File: cves/CVE-2011-3965.yml (hunk 4 above)
Comment (hkdahal) [reply cves/CVE-2011-3965.yml]: fixed, thanks!

File: cves/CVE-2016-1640.yml (hunk 5 above)
Comment (hkdahal) [reply cves/CVE-2016-1640.yml]: just added. thanks

File: cves/CVE-2016-1640.yml (hunk 6 above)
Comment (hkdahal) [reply cves/CVE-2016-1640.yml]: fixed, thanks!

File: cves/CVE-2011-3965.yml [hunk 8]
```diff
@@ -183,8 +187,9 @@ lessons:
     note:
   distrust_input:
     applies: true
-    note: Checking and ensuring that the data that comes from outside the system
-    is valid before manipulating them could go a long way.
+    note: |
+      Checking and ensuring that the data that comes from outside the system
+      is valid before manipulating them could go a long way.
```
[the remark this answers is not in the document]
Comment (hkdahal) [reply cves/CVE-2011-3965.yml]: good catch, thanks!

File: cves/CVE-2016-1640.yml (hunk 7 above)
Comment (hkdahal) [reply cves/CVE-2016-1640.yml]: thanks!

**Pull request merged**